Yoast/javascript
# Implement functionality to determine a unique stem for German words

## Summary

This PR can be summarized in the following changelog entry:

* [non-user-facing] Creates unique stems for German words.

## Relevant technical choices:
* The functionality introduced here isn't used to create German word forms, but intended to be used for internal linking.
* For words that are on exception lists in multiple categories, we cannot currently determine the correct stem. E.g. "schwamm" could either be the noun "Schwamm" ("sponge") or the past tense form "schwamm" of the verb "schwimmen" (to swim).

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test` to confirm that the coverage of the added functionality is 100%.
* Add additional German words to the spec and check whether they get stemmed correctly. Specifically, when you add different forms of a same word, they should receive the same stem.

Fixes https://github.com/Yoast/YoastSEO.js/issues/2203
